### PR TITLE
DOC: Add notes to ``rng.bytes()``

### PR DESCRIPTION
**Pull request by linus-md**
This describes hot the random bytes are generated.

### CODE REVIEW

File: numpy/random/_generator.pyx [hunk 1]
```diff
@@ -717,6 +717,14 @@ cdef class Generator:
         out : bytes
             String of length `length`.
 
+        Notes
+        -----
+        This function generates random integers from a discrete uniform
+        distribution in the range [0, 2**32), converts them to little-endian
+        byte arrays, and then decodes the resulting byte array using the 
+        UTF-32 codec. The generated integers are interpreted as 32-bit
```
Comment (rkern) [inline numpy/random/_generator.pyx:725]: There are a few factual details here that are wrong, but before correcting them, I'm not really sure that this is something that needs to be documented.

Comment (linus-md) [reply numpy/random/_generator.pyx:725]: As this is a public function I would say it deserves some comment on how it works. There should at least be a comment on the uniformity of the distribution.

Comment (rkern) [reply numpy/random/_generator.pyx:725]: I would confine the documentation to the properties of the results (e.g. uniformity, independence from the CPU's native endianness), not the step-by-step details of the implementation. We generally do not document such implementation details.

File: numpy/random/_generator.pyx [hunk 2]
```diff
@@ -717,6 +717,12 @@ cdef class Generator:
         out : bytes
             String of length `length`.
 
+        Notes
+        -----
+        This function generates random integers from a discrete uniform
+        distribution. It is independent of the from the CPU's native
```
Comment (mdhaber) [inline numpy/random/_generator.pyx:723]:
```suggestion
        distribution. The results are independent of the CPU's native
```
As written, "it" would refer to "this function", but I don't think that's what's intended.

We need to pick either "of the" or "from the", and I think "of the" is correct.

Comment (mdhaber) [reply numpy/random/_generator.pyx:723]: But this still doesn't describe what the function does if the output is a string.

File: numpy/random/_generator.pyx [hunk 3]
```diff
@@ -717,6 +717,12 @@ cdef class Generator:
         out : bytes
             String of length `length`.
 
+        Notes
+        -----
+        This function generates random integers from a discrete uniform
```
Comment (mdhaber) [inline numpy/random/_generator.pyx:722]: If this is the way to describe what the function does, we'd need to specify the domain of this distribution of random integers.

But maybe this is not the way to describe what the function does, and instead this should focus on properties of the output bytes.

Comment (linus-md) [reply numpy/random/_generator.pyx:722]: I am sorry I didn't mean to write integers. I will fix it over the weekend.

**Pull request merged**